harness/determined/keras/_tf_keras_trial.py

```diff
@@ -650,6 +650,16 @@ def run(self) -> None:
             self._launch_fit()
         except det.errors.WorkerFinishedGracefully:
             pass
+        except det.InvalidHP as e:
+            for _, _, response_func in self.workloads:
+                logging.info("Invalid hyperparameter exception in trial: {}".format(e))
+                response_func(
+                    det.util.wrap_metrics(
+                        {},
+                        self.context.get_stop_requested(),
+                        True,
+                    )
+                )
```

Comment:
Hm... this is maybe not going to do quite what you are thinking.  this hack of reading from the workloads is only valid before any workloads have been read (like when you initialize the trial), but I don't see how you can guarantee that if you are catching InvalidHP exceptions from the entire fit loop.

Reply:
@rb-determined-ai Then I should probably take a similar approach to `_pytorch_trial.py` and catch exceptions in this area of the code: https://github.com/determined-ai/determined/blob/master/harness/determined/keras/_tf_keras_trial.py#L778 for both keras and estimators?

Reply:
Hm, yes I think so.   It will be a bit trickier though, because the places where user code is called are a bit more numerous, due the the Keras Callbacks support.